contracts/UniswapSlidingWindowOracle.sol

```diff
@@ -0,0 +1,111 @@
+pragma solidity =0.6.6;
+pragma experimental ABIEncoderV2;
+
+import "./UniswapContracts.sol";
+
+// sliding window oracle that uses observations collected over a window to provide moving price averages in the past
+// `windowSize` with a precision of `windowSize / granularity`
+// note this is a singleton oracle and only needs to be deployed once per desired parameters, which
+// differs from the simple oracle which must be deployed once per pair.
+contract UniswapSlidingWindowOracle {
+    using FixedPoint for *;
+    using SafeMath for uint;
+
+    struct Observation {
+        uint timestamp;
+        uint price0Cumulative;
+        uint price1Cumulative;
+    }
+
+    // the desired amount of time over which the moving average should be computed, e.g. 24 hours
+    uint public immutable windowSize;
+    // the number of observations stored for each pair, i.e. how many price observations are stored for the window.
+    // as granularity increases from 1, more frequent updates are needed, but moving averages become more precise.
+    // averages are computed over intervals with sizes in the range:
+    //   [windowSize - (windowSize / granularity) * 2, windowSize]
+    // e.g. if the window size is 24 hours, and the granularity is 24, the oracle will return the average price for
+    //   the period:
+    //   [now - [22 hours, 24 hours], now]
```

Comment:
i dont really understand this comment

Reply:
I think it explains this line https://github.com/compound-finance/open-oracle/pull/74/files/3139b366529da7f4a0c20391159090852683b230#diff-3d75c7504e865fedc968a253f45deae0R92
It was inherited from Uniswap example, it should go most likely together with that ^ weird check.